design-documents/graph-ql/coverage/gift-registry.graphqls

```diff
@@ -0,0 +1,296 @@
+}
+
+type GiftRegistrySelectAttributeMetadata implements GiftRegistryDynamicAttributeMetadataInterface, GiftRegistrySelectAttributeMetadataInterface {
+}
+
+type GiftRegistryDateAttributeMetadata implements GiftRegistryDynamicAttributeMetadataInterface {
+    format: GiftRegistryDateAttributeFormat!
+}
+
+enum GiftRegistryDateAttributeFormat {
+    SHORT
+    MEDIUM
+    LONG
+    FULL
+}
+
+type GiftRegistryCountryAttributeMetadata implements GiftRegistryDynamicAttributeMetadataInterface {
+    show_region: Boolean!
+}
+
+type GiftRegistryEventCountryAttributeMetadata implements GiftRegistryDynamicAttributeMetadataInterface, GiftRegistrySearcheableAttributeMetadataInterface {
+    show_region: Boolean!
+}
+
+type GiftRegistryEventDateAttributeMetadata implements GiftRegistryDynamicAttributeMetadataInterface, GiftRegistrySearcheableAttributeMetadataInterface {
+    format: GiftRegistryDateAttributeFormat!
+}
+
+type GiftRegistryEventLocationAttributeMetadata implements GiftRegistryDynamicAttributeMetadataInterface, GiftRegistrySearcheableAttributeMetadataInterface {
+}
+
+type GiftRegistryRoleAttributeMetadata implements GiftRegistryDynamicAttributeMetadataInterface, GiftRegistrySearcheableAttributeMetadataInterface, GiftRegistrySelectAttributeMetadataInterface {
+}
+
+interface GiftRegistrySearcheableAttributeMetadataInterface {
+    is_searcheable: Boolean!
+}
+
+interface GiftRegistrySelectAttributeMetadataInterface {
+    options: [GiftRegistrySelectAttributeOptionMetadata]
+}
+
+type GiftRegistrySelectAttributeOptionMetadata {
+    code: ID!
+    label: String!
+    is_default: Boolean
+}
+
+type GiftRegistry {
+    id: ID!
+    event_name: String!
+    type: GiftRegistryType
+    message: String!
+    created_on: String! @doc(description: "Creation date") @doc(description: "Accessible to the registry owner only")
+    privacy_settings: GiftRegistryPrivacySettings! @doc(description: "Accessible to the registry owner only")
+    status: GiftRegistryStatus! @doc(description: "Accessible to the registry owner only")
+    owner_name: String!
+    registrants: [GiftRegistryRegistrant]
+    shipping_address: CustomerAddress @doc(description: "Accessible to the registry owner only")
+    dynamic_attributes: [GiftRegistryDynamicAttribute]
+    items: [GiftRegistryItemInterface]
+}
+
+type GiftRegistryItemInterface {
+    id: String!
+    quantity: Float!
+    quantity_fulfilled: Float!
+    note: String
+    added_on: String!
+    product: ProductInterface
+}
+
+type SimpleGiftRegistryItem implements GiftRegistryItemInterface {
+    customizable_options: [SelectedCustomizableOption]
+}
+
+type BundleGiftRegistryItem implements GiftRegistryItemInterface {
+    customizable_options: [SelectedCustomizableOption]
+    bundle_options: [SelectedBundleOption!]
+}
+
+type ConfigurableGiftRegistryItem implements GiftRegistryItemInterface {
+    customizable_options: [SelectedCustomizableOption]
```

Comment:
in case if configurable options were set up should we point the particular SKU?

Reply:
See definition of `GiftRegistryItemInterface`, it has reference to product. Do we need something else?

Reply:
@paliarush this is slightly unclear to me. For bundles and configurable, this query should return the selected simples, because these products assuming to be added to cart, right? But as far I understood a product in the gift registry interface could be bundle or configurable. 

What will be the flow of adding the item from the registry to the shopping cart?

My thoughts were that gift registry should provide explicit arguments that our add to cart mutation requires?

Reply:
There should be a way to navigate from the registry to the configurable/bundle product page. How can this work if we only return simple product data?

Also, existing behavior is aligned with cart behavior. And there should be enough information in gift registry item data to add it to cart. The exact queries will be added soon in the use cases section of this PR.